src/build.rs

```diff
@@ -171,6 +171,28 @@ impl<'cb> RepoBuilder<'cb> {
         self
     }
 
+    /// Specify fetch depth, a value less than zero is interpreted as pull everything (effectively the same as no declaring a limit depth)
+    pub fn depth(&mut self, depth: i32) -> &mut RepoBuilder<'cb> {
```

Comment:
Instead of adding a new method here, would it be possible to leave this as a setting that required calling `.fetch_options(...)` instead? I am reluctant to try to repeat low-level option settings, since that would require deciding which settings are or are not replicated, and possibly causing confusion when something could be specified in multiple places.

Reply:
So this should be deleted? I find the `depth` parameter to be kinda (very) important imo, but it can be removed without any problems.